### PR TITLE
FIX/WIKI-1438:Permissions in Wiki settings global does not work

### PR DESCRIPTION
**Pull request by lbengayed**
This is a proposed fix. It solves the issue WIKI-1438.
With this PR, When the user creates a new page, which inherits from a permission updated page, the new wiki page will inherits the permissions of the mother wiki page.
Also, When the space manager updates the permissions in the Wiki global permissions, all pages permissions will be updated, except for wiki pages that was changed from the permissions button in that page action pulldown.

### CODE REVIEW

**Comment by lbengayed** [comment]
The Wiki global settings does not work.
This fix settled the Wiki global settings  for all wiki pages.

File: wiki-webui/src/main/java/org/exoplatform/wiki/webui/control/filter/EditPagesPermissionFilter.java [hunk 1]
```diff
@@ -42,11 +44,18 @@ public EditPagesPermissionFilter(String messageKey) {
...
   public boolean accept(Map<String, Object> context) throws Exception {
+    Wiki wiki = Utils.getCurrentWiki();
     Page page = Utils.getCurrentWikiPage();
-    if (page == null) {
+    Identity currentIdentity = ConversationState.getCurrent().getIdentity();
+    if (wiki == null) {
       return false;
     }
-    return wikiService.hasPermissionOnPage(page, PermissionType.EDITPAGE, ConversationState.getCurrent().getIdentity());
+    else if (wikiService.hasPermissionOnWiki(wiki, PermissionType.EDITPAGE, currentIdentity) && wikiService.hasPermissionOnPage(page, PermissionType.EDITPAGE, currentIdentity))
```
Comment (thomasdelhomenie) [inline wiki-webui/src/main/java/org/exoplatform/wiki/webui/control/filter/EditPagesPermissionFilter.java]: If the user has the edit permissions on the page but not on the wiki, it should be authorized to edit the page according to the spec, whereas it is not the case with this condition.

File: wiki-webui/src/main/java/org/exoplatform/wiki/webui/control/filter/EditPagesPermissionFilter.java [hunk 2]
```diff
@@ -42,11 +44,18 @@ public EditPagesPermissionFilter(String messageKey) {
...
+    {
+      return wikiService.hasPermissionOnWiki(wiki, PermissionType.EDITPAGE, currentIdentity);
+    }
+
+    return false;
```
Comment (thomasdelhomenie) [inline wiki-webui/src/main/java/org/exoplatform/wiki/webui/control/filter/EditPagesPermissionFilter.java]: IMO the permission logic should be implemented in the service layer (WikiService.hasPermissionOnPage).

File: wiki-webui/src/main/java/org/exoplatform/wiki/webui/control/filter/EditPagesPermissionFilter.java [hunk 3]
```diff
@@ -42,11 +44,18 @@ public EditPagesPermissionFilter(String messageKey) {
   
   @Override
   public boolean accept(Map<String, Object> context) throws Exception {
+    Wiki wiki = Utils.getCurrentWiki();
     Page page = Utils.getCurrentWikiPage();
-    if (page == null) {
+    Identity currentIdentity = ConversationState.getCurrent().getIdentity();
+    if (wiki == null) {
```
Comment (thomasdelhomenie) [inline wiki-webui/src/main/java/org/exoplatform/wiki/webui/control/filter/EditPagesPermissionFilter.java]: Shouldn't we keep the nullity check on the page as well ?

**Comment by plamarque** [comment]
@ahamdi could you review this PR. Ghis ticket has been pending for too long.

**Pull request closed without merging**